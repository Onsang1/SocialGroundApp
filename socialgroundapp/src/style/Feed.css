.feed-header {
    margin-top: 20px;
    margin-bottom: 70px;
    display: flex;
    justify-content: flex-start;
    flex-direction: row;
    align-items: center;
}

.navbar {
    margin-top: 20px;
    margin-bottom: 70px;
    display: flex;
    justify-content: flex-start;
    flex-direction: row;
    align-items: center;
    border-bottom: 2px solid #dfdededa;
    padding-bottom: 8px;
}

.left-section-feed {
    display: inline-block;
    width: 300px;
    margin-left: 15px;
}

.right-section-feed {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-end;
    flex: 1;
    margin-right: 100px;
    flex-shrink: 0;
    margin-left: 50%;
}

.search-icon-area,
.plus-sign-area,
.profile-icon-area {
    display: inline-block;
    height: 40px;
    object-fit: contain;
    width: 40px;
}



.small-logo {
    width: 200px;
}

.search-icon {
    height: 35px;
    width: 35px;
}

.navbar button {
    background-color: white;
    border: none;
    height: 35px;
    width: 35px;
    padding: 0;
    margin: 0;
}

.plus-sign-icon {
    height: 35px;
    width: 35px;
}

.profile-icon {
    height: 35px;
    width: 35px;
}

#posts {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
}

.post {

    margin-top: 20px;
    margin-bottom: 100px;
    border-radius: 8px;
    display: flex;
    flex-direction: column;
    width: 500px;
    margin-right: auto;
    margin-left: auto;
    border-style: solid;
    border-color: rgb(217, 214, 214);
    border-width: 2px;
    overflow: hidden;
    opacity: 1;
}

.profile-info {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
}

.profile-area {
    display: inline-block;
    width: 45px;
    margin: 5px;
}


#post-profile {
    margin-left: 5px;
    margin-top: 0;
    width: 35px;
    height: 35px;
}

.post-image {
    display: block;
    object-fit: contain;
    width: 100%;
}

.like-box {
    display: inline-block;
    width: 35px;
    margin-right: 5px;
}

.comment-box {
    display: inline-block;
    width: 35px;
}

.numComments {
    margin-left: 5px;
}

.like-icon {
    padding-top: 5px;
    height: 30px;
    width: 30px;
    padding-left: 5px;
}

.comment-icon {
    padding-top: 5px;
    height: 30px;
    width: 30px;
}

.post-box {
    display: flex;
}

.post-box .username {
    margin-left: 5px;
    margin-top: 0;
    font-size: 20px;
    color: black;
    font-weight: bold;
    display: inline-block;
    margin-right: 5px;
    text-decoration: none;
}

.caption {
    font-size: 18px;
    color: black;
    font-weight: 300;
    vertical-align: top;
    display: inline-block;
    margin: 0%;
    padding-top: 2px;
    overflow-wrap: break-word;
}

.captionArea {
    height: 35px;
    display: flex;
    align-items: center;
    justify-content: flex-start;
}

.smily-face {
    display: inline-block;
    width: 45px;
    margin-left: 5px;
}

.happy-face-icon {
    object-fit: contain;
    width: 40px;
    padding-top: 8px;
}

.form-input {
    display: flex;
    margin-top: 10px;
    justify-content: center;
    align-items: center;
    resize: none;
    overflow-y: auto;
}

.post-action {
    display: inline-block;
    width: 45px;
    margin-left: 5px;
    vertical-align: baseline;
}

.post-btn {
    background-color: rgb(10, 182, 240);
    border-radius: 5px;
    border-color: blue;
    margin-bottom: 5px;
}

.form-entry-box {
    justify-content: left;
    display: flex;
}

.post-input {
    flex: 1;
}

.comment-bar {
    padding-top: 5px;
    border-radius: 5px;
    border-color: lightgrey;
    border-style: solid;
    background: #fafafa;
    opacity: 3;
    margin-left: 2px;
    width: 260px;
    height: 27px;
    font-size: 22px;
}

.bottom {
    margin-bottom: 10px;
}