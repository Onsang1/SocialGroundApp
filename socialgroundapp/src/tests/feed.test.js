import React from 'react';
// import testing library functions
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import {within} from '@testing-library/dom'
import Feed from "../components/ActivityFeed"
import CreateComment from '../components/Create';


test('renders Send Button', () => {
  const { getByText } = render(<CreateComment />);
  const linkElement = getByText('SEND');
  expect(linkElement).toBeInTheDocument();
});


test('renders username mario', () => {
  const { getByText } = render(<Post />);
  const linkElement = getByText('mario');
  expect(linkElement).toBeInTheDocument();
});

test('renders username yoshi', () => {
  const { getByText } = render(<Post />);
  const linkElement = getByText('yoshi');
  expect(linkElement).toBeInTheDocument();
});

test('renders username yoshi', () => {
  const { getByText } = render(<Comment />);
  const linkElement = getByText('username');
  expect(linkElement).toBeInTheDocument();
});

test('feed matches snapshot', () => {
  const component = renderer.create(
    <Router>
      <Feed />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('comment matches snapshot', () => {
  const component = renderer.create(
    <Router>
      <Comment />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('Textbox to have value after input', async () => {
  render(
    <Router>
      <CreateComment />
    </Router>,
  );

  const commentLabel = getByText('Type');
  const commentPlaceholder = getByText('Add a comment');

  expect(commentLabel).toBeInDocument();
  expect(commentPlaceholder).toBeInDocument();

  const commentInput = getByLabelText('Add a comment');

  expect(commentInput).toHaveAttribute('maxRows', 'placeholder','required','variant');

  const element = screen.getByRole('textbox', { name: 'Add a comment' });

  await userEvent.type(element, 'username123@gmail.com');
  expect(screen.getByLabelText('Add a comment')).toHaveValue('user123@gmail.com');
});
