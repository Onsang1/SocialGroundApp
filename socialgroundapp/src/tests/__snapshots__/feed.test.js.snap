// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comment matches snapshot 1`] = `
<ul
  className="MuiList-root makeStyles-root-9 MuiList-padding"
>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
    disabled={false}
  >
    <div
      className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular"
      >
        <img
          alt="Remy Sharp"
          className="MuiAvatar-img"
          src="https://images.unsplash.com/photo-1609220361638-14ceb45e5e1e?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max"
        />
      </div>
    </div>
    <div
      className="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        username
      </span>
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        <span
          className="MuiTypography-root makeStyles-inline-10 MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          this is comment
        </span>
      </p>
    </div>
  </li>
  <li
    className="MuiDivider-root MuiDivider-inset"
    role="separator"
  />
</ul>
`;

exports[`feed matches snapshot 1`] = `
<div
  className="Feed"
>
  <nav
    className="navbar"
  >
    <div
      className="left-section-feed"
    >
      <img
        alt="social ground logo "
        className="small-logo"
        src={Object {}}
      />
    </div>
    <div
      className="right-section-feed"
    >
      <div
        className="search-icon-area"
      >
        <a
          href="/Find"
          onClick={[Function]}
        >
          <button
            className="icon-btn"
          >
            <img
              alt="search icon"
              className="search-icon"
              src={Object {}}
            />
          </button>
        </a>
      </div>
      <div
        className="plus-sign-area"
      >
        <a
          href="/upload"
          onClick={[Function]}
        >
          <button
            className="plus-btn"
          >
            <img
              alt="plus sign icon"
              className="plus-sign-icon"
              src={Object {}}
            />
          </button>
        </a>
      </div>
      <div
        className="profile-icon-area"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          <button
            className="profile-btn"
          >
            <img
              alt="profile-icon"
              className="profile-icon"
              src={Object {}}
            />
          </button>
        </a>
      </div>
    </div>
  </nav>
  <section>
    <div>
      Loading...
    </div>
  </section>
</div>
`;
